feat(frontend): created the login page

Mahmoud-Hanyy opened the pull request:
## 📌 Summary

Created the Login page

## ✅ What Was Done

- [x] Developed the login form
- [x] Made the footer
- [x] Adjusted the global styling
- [x] Installed packages for styling and icons
- [x] Used the logo image from the assets

## 🧪 How to Test

<!-- How should a reviewer test this locally? Be as specific as possible -->

1. Run `docker compose up`
2. Open `http://localhost:3000`
3. Login as demo user (email: ..., password: ...)
4. Confirm that feature X behaves as expected
5. Run tests: `docker compose exec web pytest`


## 🔗 Related Issues

Closes #2

## 📋 Checklist Before Merge

- [x] Code is self-documented and clean
- [x] Tests are added/updated
- [x] Docs/README updated if necessary
- [x] I rebased onto latest `dev` and resolved any conflicts

Mahmoud-Hanyy committed:
feat(frontend): created the login page

Created the Login page
- developed the login form
- made the footer
- adjusted the global styling
- installed packages for styling and icons
- used the logo image from the assets

kerolosNabil247 commented on frontend/src/index.css:
```diff
@@ -1,14 +1,39 @@
+  background-color: var(--primary-color);
+  margin: 0;
+  padding: 0;
+}
```
You can add these css styles as tailwind classes

kerolosNabil247 commented on frontend/src/index.css:
```diff
@@ -1,14 +1,39 @@
 
+.forget-password:hover{
+  color: white;
+}
```
We don't need this style as it is not mentioned in the design

kerolosNabil247 commented on frontend/src/index.css:
```diff
@@ -1,14 +1,39 @@
+  color: var(--secondary-color);
+  font-weight: semibold;
+  font-size: 18px;
+}
```
You can also add these css styles as tailwind classes

kerolosNabil247 reviewed (changes requested):
Great job, Mahmoud, but there are some minor changes.

Mahmoud-Hanyy committed:
fix(frontend): Fix the login page bugs

Fixed some bugs in the login page:
- Deleted unwanted styles in css file
- Replaced the styles in index.css to tailwind classes
Closes #2

kerolosNabil247 reviewed (approved):
Great job, Mahmoud on fixing the code
